dot/build_spec_integration_test.go

```diff
@@ -0,0 +1,165 @@
+//go:build integration
+// +build integration
+
+// Copyright 2020 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package dot
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuildFromGenesis(t *testing.T) {
+	t.Parallel()
+
+	file, err := genesis.CreateTestGenesisJSONFile(false)
+	defer os.Remove(file)
+	require.NoError(t, err)
+	bs, err := BuildFromGenesis(file, 0)
+
+	expectedChainType := "TESTCHAINTYPE"
+	expectedProperties := map[string]interface{}{
+		"ss58Format":    0.0,
+		"tokenDecimals": 0.0,
+		"tokenSymbol":   "TEST",
+	}
+
+	bs.genesis.ChainType = expectedChainType
+	bs.genesis.Properties = expectedProperties
+
+	require.NoError(t, err)
+
+	// confirm human-readable fields
+	hr, err := bs.ToJSON()
+	require.NoError(t, err)
+	jGen := genesis.Genesis{}
+	err = json.Unmarshal(hr, &jGen)
+	require.NoError(t, err)
+	genesis.TestGenesis.Genesis = genesis.TestFieldsHR
+	require.Equal(t, genesis.TestGenesis.Genesis.Runtime, jGen.Genesis.Runtime)
+	require.Equal(t, expectedChainType, jGen.ChainType)
+	require.Equal(t, expectedProperties, jGen.Properties)
+
+	// confirm raw fields
+	raw, err := bs.ToJSONRaw()
+	require.NoError(t, err)
+	jGenRaw := genesis.Genesis{}
+	err = json.Unmarshal(raw, &jGenRaw)
+	require.NoError(t, err)
+	genesis.TestGenesis.Genesis = genesis.TestFieldsRaw
+	require.Equal(t, genesis.TestGenesis.Genesis.Raw, jGenRaw.Genesis.Raw)
+	require.Equal(t, expectedChainType, jGenRaw.ChainType)
+	require.Equal(t, expectedProperties, jGenRaw.Properties)
+}
+
+func TestBuildFromGenesis_WhenGenesisDoesNotExists(t *testing.T) {
+	t.Parallel()
+
+	bs, err := BuildFromGenesis("/not/exists/genesis.json", 0)
+	require.Nil(t, bs)
+	require.Error(t, err, os.ErrNotExist)
+}
+
+func TestWriteGenesisSpecFileWhenFileAlreadyExists(t *testing.T) {
+	t.Parallel()
+
+	f, err := ioutil.TempFile("", "existing file data")
+	require.NoError(t, err)
+	defer os.Remove(f.Name())
+
+	someBytes := []byte("Testing some bytes")
+	err = WriteGenesisSpecFile(someBytes, f.Name())
+
+	require.Error(t, err,
+		fmt.Sprintf("file %s already exists, rename to avoid overwritten", f.Name()))
```

Comment:
`require.Error` only checks the error is not nil

```suggestion
	require.EqualError(t, err,
		fmt.Sprintf("file %s already exists, rename to avoid overwritten", f.Name()))
```